cmd/otel-allocator/watcher/promOperator.go

```diff
@@ -139,6 +196,48 @@ func (w *PrometheusCRWatcher) Watch(upstreamEvents chan Event, upstreamErrors ch
 	// this channel needs to be buffered because notifications are asynchronous and neither producers nor consumers wait
 	notifyEvents := make(chan struct{}, 1)
 
+	if w.nsInformer != nil {
+		go w.nsInformer.Run(w.stopChannel)
+		if ok := cache.WaitForNamedCacheSync("namespace", w.stopChannel, w.nsInformer.HasSynced); !ok {
+			success = false
+		}
+
+		_, _ = w.nsInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(oldObj, newObj interface{}) {
```

Comment:
I don't see a test for this logic. Can we add one?

Reply:
@swiatekm-sumo - Added this too. Please take a look.